sdk/spring/azure-spring-boot/src/main/java/com/azure/spring/aad/webapi/AADResourceServerOAuth2AuthorizedClientRepository.java

```diff
@@ -42,44 +44,65 @@
 
 /**
  * <p>
- * OAuth2AuthorizedClientRepository for obo flow
+ * AADResourceServerOAuth2AuthorizedClientRepository for obo flow and client credential flow
  * </p>
  */
-public class AADOAuth2OboAuthorizedClientRepository implements OAuth2AuthorizedClientRepository {
+public class AADResourceServerOAuth2AuthorizedClientRepository implements OAuth2AuthorizedClientRepository {
 
-    private static final Logger LOGGER = LoggerFactory.getLogger(AADOAuth2OboAuthorizedClientRepository.class);
+    private static final Logger LOGGER =
+        LoggerFactory.getLogger(AADResourceServerOAuth2AuthorizedClientRepository.class);
 
     private static final String OBO_AUTHORIZEDCLIENT_PREFIX = "obo_authorizedclient_";
 
     private final ClientRegistrationRepository repository;
 
-    public AADOAuth2OboAuthorizedClientRepository(ClientRegistrationRepository repository) {
+    private final OAuth2AuthorizedClientService oAuth2AuthorizedClientService;
+
+    public AADResourceServerOAuth2AuthorizedClientRepository(ClientRegistrationRepository repository) {
+        this(null, repository);
+    }
+
+    public AADResourceServerOAuth2AuthorizedClientRepository(OAuth2AuthorizedClientService oAuth2AuthorizedClientService,
+                                                             ClientRegistrationRepository repository) {
         this.repository = repository;
+        this.oAuth2AuthorizedClientService = oAuth2AuthorizedClientService;
     }
 
     @Override
     @SuppressWarnings({ "unchecked", "rawtypes" })
     public <T extends OAuth2AuthorizedClient> T loadAuthorizedClient(String registrationId,
-                                                                     Authentication authentication,
+                                                                     Authentication principal,
                                                                      HttpServletRequest request) {
+        ClientRegistration clientRegistration = repository.findByRegistrationId(registrationId);
+        if (clientRegistration == null) {
+            LOGGER.error("Not found the ClientRegistration, registrationId={}", registrationId);
+            return null;
+        }
+        if (clientRegistration.getAuthorizationGrantType().getValue().equals(
```

Comment:
we could add a function to the enum to check whether the AADAuthorizationGrantType and AuthrizationGrantType are equal.